### PR TITLE
Axios upgrade

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Fixes axios security alert

## Discussion

- Most of the pain upgrading from 0.27.2 was eliminated by a fix in 0.28.1

## Checklist

- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

File: src/contrib/google/sheets/core/handleGoogleRequestRejection.ts [hunk 1]
```diff
@@ -43,9 +40,13 @@ export async function handleGoogleRequestRejection(
 ): Promise<Error> {
   // Request errors from proxyRequest are wrapped in ContextError which includes metadata about the integration
   // configuration. Therefore, get root cause for determining if this is an Axios error
-  const rootCause = getRootCause(error);
+  const rootCause = selectAxiosError(error);
 
-  console.debug("Error making Google request", { error });
+  console.log({ rootCause: JSON.stringify(rootCause, null, 2) });
```
Comment (fungairino) [inline src/contrib/google/sheets/core/handleGoogleRequestRejection.ts:45]: maybe we should just include this detail in the console.debug statement below?

File: src/errors/errorHelpers.ts [hunk 2]
```diff
@@ -144,6 +144,7 @@ export function selectSpecificError<
  */
 export function getRootCause(error: unknown): unknown {
   while (isErrorObject(error) && error.cause) {
+    console.log("get root cause", error);
```
Comment (fungairino) [inline src/errors/errorHelpers.ts:147]: Intentional to leave this here?

**Review by fungairino — APPROVED** [review]

File: src/errors/errorHelpers.ts (hunk 2 above)
Comment (grahamlangford) [reply src/errors/errorHelpers.ts]: Nope, you just got here before I did

File: src/contrib/google/sheets/core/handleGoogleRequestRejection.ts (hunk 1 above)
Comment (grahamlangford) [reply src/contrib/google/sheets/core/handleGoogleRequestRejection.ts]: Wasn't intended to be in the PR. Just aiding readability of errors when running tests in the terminal.

**Review by BLoe — APPROVED** [review]

File: src/contrib/google/sheets/core/handleGoogleRequestRejection.ts [hunk 3]
```diff
@@ -54,12 +48,17 @@ export async function handleGoogleRequestRejection(
     });
   }
 
-  if (!isAxiosError(rootCause) || rootCause.response == null) {
+  // Request errors from proxyRequest are wrapped in ContextError which includes metadata about the integration
+  // configuration. Therefore, get axios error if it exists.
+  const axiosError = selectAxiosError(error);
+
+  if (axiosError.response == null) {
```
Comment (BLoe) [inline src/contrib/google/sheets/core/handleGoogleRequestRejection.ts:55]:
```suggestion
  if (axiosError?.response == null) {
```

Comment (BLoe) [reply src/contrib/google/sheets/core/handleGoogleRequestRejection.ts:55]: Might be able to remove the disable-lint hints below if you fix this?

`selectAxiosError` can return `null`

Comment (grahamlangford) [reply src/contrib/google/sheets/core/handleGoogleRequestRejection.ts]: That's weird. I wonder why typescript was passing for me locally before. But you're right. Fixed.

**Pull request merged**